services/app-web/src/components/Modal/UnlockSubmitModal.tsx

```diff
@@ -0,0 +1,202 @@
+import { Modal } from './Modal'
+import styles from './UnlockSubmitModal.module.scss'
+import { PoliteErrorMessage } from '../PoliteErrorMessage'
+import * as Yup from 'yup'
+
+type ModalType = 'SUBMIT' | 'RESUBMIT' | 'UNLOCK'
+
+type ModalValueType = {
+    modalHeading?: string
+    onSubmitText?: string
+    modalDescription?: string
+    inputHint?: string
+    unlockSubmitModalInputValidation?: string
+}
+
+const modalValueDictionary: { [Property in ModalType]: ModalValueType } = {
+    RESUBMIT: {
+        modalHeading: 'Summarize changes',
+        onSubmitText: 'Resubmit',
+        modalDescription:
+            'Once you submit, this package will be sent to CMS for review and you will no longer be able to make changes.',
+        inputHint: 'Provide summary of all changes made to this submission',
+        unlockSubmitModalInputValidation:
+            'You must provide a summary of changes',
+    },
+    UNLOCK: {
+        modalHeading: 'Reason for unlocking submission',
+        onSubmitText: 'Unlock',
+        inputHint: 'Provide reason for unlocking',
+        unlockSubmitModalInputValidation:
+            'You must provide a reason for unlocking this submission',
+    },
+    SUBMIT: {
+        modalHeading: 'Ready to submit?',
+        onSubmitText: 'Submit',
+        modalDescription:
+            'Submitting this package will send it to CMS to begin their review.',
+    },
+}
+
+export const UnlockSubmitModal = ({
+    healthPlanPackage,
+    submissionName,
+    modalType,
+    modalRef,
+    setIsSubmitting,
+}: {
+    healthPlanPackage: UnlockedHealthPlanFormDataType | HealthPlanPackage
+    submissionName?: string
+    modalType: ModalType
+    modalRef: React.RefObject<ModalRef>
+    setIsSubmitting?: React.Dispatch<React.SetStateAction<boolean>>
+}): React.ReactElement => {
+    const [focusErrorsInModal, setFocusErrorsInModal] = useState(true)
+    const [modalAlert, setModalAlert] = useState<string | undefined>(undefined) // when api errors error
+    const navigate = useNavigate()
+    const modalValues: ModalValueType = modalValueDictionary[modalType]
+
+    const modalFormInitialValues = {
+        unlockSubmitModalInput: '',
+    }
+
+    const [submitHealthPlanPackage, { loading: submitMutationLoading }] =
+        useSubmitHealthPlanPackageMutation()
+    const [unlockHealthPlanPackage, { loading: unlockMutationLoading }] =
+        useUnlockHealthPlanPackageMutation()
+
+    const formik = useFormik({
+        initialValues: modalFormInitialValues,
+        validationSchema: Yup.object().shape({
+            unlockSubmitModalInput: Yup.string().defined(
+                modalValues.unlockSubmitModalInputValidation
+            ),
+        }),
+        onSubmit: (values) => onSubmit(values.unlockSubmitModalInput),
+    })
+
+    const mutationLoading =
+        modalType === 'UNLOCK' ? unlockMutationLoading : submitMutationLoading
+    const isSubmitting = mutationLoading || formik.isSubmitting
+
```

Comment:
`const includesFormInput =   modalType === 'UNLOCK' || modalType === 'RESUBMIT'`
seems like another shared variable in this refactor that might help make it easier to grok. That logic is used both to determine submitHandler and to determine whether or not to add the form